diff --git a/p4-fusion/CMakeLists.txt b/p4-fusion/CMakeLists.txt
index 7fc861e..efb7c4d 100644
--- a/p4-fusion/CMakeLists.txt
+++ b/p4-fusion/CMakeLists.txt
@@ -5,8 +5,8 @@ if (NOT OPENSSL_ROOT_DIR)
     set(OPENSSL_ROOT_DIR /usr/local/ssl)
 endif()
 
-set(OPENSSL_USE_STATIC_LIBS true)
-find_package(OpenSSL)
+find_package(PkgConfig)
+pkg_check_modules(OPENSSL openssl)
 
 add_executable(p4-fusion ${Headers} ${Sources})
 
@@ -37,17 +37,13 @@ if (APPLE)
     )
 endif(APPLE)
 
-message(${OPENSSL_SSL_LIBRARIES})
-message(${OPENSSL_CRYPTO_LIBRARIES})
-
 target_link_libraries(p4-fusion PUBLIC
     ${Frameworks}
     client
     rpc
     supp
     p4script_cstub
-    ${OPENSSL_SSL_LIBRARIES}
-    ${OPENSSL_CRYPTO_LIBRARIES}
+    ${OPENSSL_LINK_LIBRARIES}
     git2
     minitrace
 )
